lua/which-key/state.lua

```diff
@@ -294,7 +305,8 @@ function M.start(opts)
     M.recursion = 0
   end)
 
-  if M.recursion > 50 then
+  recursion_threshold = M.hydra_mode and 200 or 50
```

Comment:
increasing the threshold for hydra mode to fix the problem # 1 as described in the pr